modules/mbedtls/CMakeLists.txt

```diff
@@ -1,71 +1,208 @@
+#
+# Copyright (c) 2023 Linaro Limited
+#
+# SPDX-License-Identifier: Apache-2.0
+#
 if(CONFIG_MBEDTLS)
-zephyr_interface_library_named(mbedTLS)
 
-if(CONFIG_MBEDTLS_BUILTIN)
-  target_compile_definitions(mbedTLS INTERFACE
-    MBEDTLS_CONFIG_FILE="${CONFIG_MBEDTLS_CFG_FILE}"
-  )
+  if(CONFIG_MBEDTLS_BUILTIN)
+    if(CONFIG_MBEDTLS_ZEPHYR_ENTROPY AND NOT CONFIG_ENTROPY_HAS_DRIVER)
+      message(WARNING "No entropy device on the system, using fake entropy source!")
+    endif()
 
-  target_include_directories(mbedTLS INTERFACE
-    ${ZEPHYR_CURRENT_MODULE_DIR}/include
-    # mbedTLS v3.1.0 has psa_crypto_cipher.c including an internal header using <>.
-    # This line ensures the header can be found in Zephyr.
-    # When updating to mbedTLS > v3.1.0, please check if this has been corrected,
-    # and if so remove this include path.
-    ${ZEPHYR_CURRENT_MODULE_DIR}/library
-    configs
-    include
-  )
+    # Library for mbed TLS includes and configurations
+    zephyr_interface_library_named(mbedTLS)
```

Comment:
This should be moved up to where it was previously so that it's included with the else() below as it was previously